src/main/java/org/ihtsdo/otf/traceabilityservice/service/TraceabilityStreamConsumer.java

```diff
@@ -47,14 +49,32 @@ public void receiveMessage(String message) throws JsonProcessingException {
 
 		final ActivityType activityType = activityMessage.getActivityType();
 		final String mergeSourceBranch = activityMessage.getSourceBranch();
-		final Activity activity = new Activity(username, branchPath, mergeSourceBranch, commitTimestamp, activityType);
+		final boolean isChangeListComplete = activityMessage.isChangeListComplete();
+
+		Activity activity = new Activity(username, branchPath, mergeSourceBranch, commitTimestamp, activityType);
+		if (!isChangeListComplete) {
+			// In this case a large amount of concept changes is sliced and queued in multiple messages for the same activity.
+			// Try to locate that activity to add the current concept changes to it.
+			Page<Activity> activities = activityRepository.findByBranchAndActivityTypeAndCommitDate
```

Comment:
@markrynbeek  Adding changes to the existing activity will make the size of the document exceed 100mb easily when processing the subsequent batches. The default value of [http.max_content_length] is set to 100MB. Elasticsearch will fail to index any documents which is bigger than 100MB.

Reply:
As discussed (based on this comment) we'll look to store multiple Activity documents instead, each with a chunk/slice of N concepts.

Reply:
This pull request can be discarded; instead of re-assembling the sliced Activity again into one, multiple of the same Activity (each with a different change list) will be added to traceability.